rpcserver.go

```diff
@@ -2446,6 +2446,13 @@ func (r *rpcServer) AddInvoice(ctx context.Context,
 	}
 
 	amt := btcutil.Amount(invoice.Value)
+	
+	// The value of the invoice must not be negative.
+	if amt < 0 {
```

Comment:
I would suggest doing this check before doing the cast to `btcutil.Amount`, right below the other checks above.